### ISSUE TITLE
SELinux on CoreOS

### ISSUE THREAD

**Issue by anshupande** [coreos/bugs#1278]
As per this document I tried,Check a container's compatibility with SELinux policy and get error message: ```May 12 19:11:00 ip-10-74-131-153.ec2.internal audit[4445]: AVC avc:  denied  { write } for  pid=4445 comm="entrypoint.sh" path="pipe:[596879]" dev="pipefs" ino=596879 scontext=system_u:system_r:svirt_lxc_net_t:s0:c845,c982 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1```

**Comment by anshupande** [comment]
```core@ip-10-74-131-153 sandbox-anshu-us-east-1c-control ~/mesos-systemd/v3/fleet-local/control $ journalctl -f|grep avc
May 12 19:30:34 ip-10-74-131-153.ec2.internal kernel: audit: type=1400 audit(1463081434.467:10249): avc:  denied  { write } for  pid=13959 comm="entrypoint.sh" path="pipe:[957490]" dev="pipefs" ino=957490 scontext=system_u:system_r:svirt_lxc_net_t:s0:c438,c919 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:34 ip-10-74-131-153.ec2.internal audit[13959]: AVC avc:  denied  { write } for  pid=13959 comm="entrypoint.sh" path="pipe:[957490]" dev="pipefs" ino=957490 scontext=system_u:system_r:svirt_lxc_net_t:s0:c438,c919 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:37 ip-10-74-131-153.ec2.internal audit[14358]: AVC avc:  denied  { write } for  pid=14358 comm="sh" path="pipe:[958332]" dev="pipefs" ino=958332 scontext=system_u:system_r:svirt_lxc_net_t:s0:c773,c997 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:38 ip-10-74-131-153.ec2.internal audit[14404]: AVC avc:  denied  { write } for  pid=14404 comm="flight-director" path="pipe:[958539]" dev="pipefs" ino=958539 scontext=system_u:system_r:svirt_lxc_net_t:s0:c106,c146 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:38 ip-10-74-131-153.ec2.internal audit[14404]: AVC avc:  denied  { ioctl } for  pid=14404 comm="flight-director" path="pipe:[958539]" dev="pipefs" ino=958539 ioctlcmd=5401 scontext=system_u:system_r:svirt_lxc_net_t:s0:c106,c146 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:41 ip-10-74-131-153.ec2.internal kernel: audit: type=1400 audit(1463081440.908:10293): avc:  denied  { write } for  pid=10492 comm="java" path="/opt/zookeeper/transactions/version-2/log.700000007" dev="xvda9" ino=14610577 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
May 12 19:30:40 ip-10-74-131-153.ec2.internal audit[10492]: AVC avc:  denied  { write } for  pid=10492 comm="java" path="/opt/zookeeper/transactions/version-2/log.700000007" dev="xvda9" ino=14610577 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
May 12 19:30:42 ip-10-74-131-153.ec2.internal audit[14780]: AVC avc:  denied  { write } for  pid=14780 comm="sh" path="pipe:[959815]" dev="pipefs" ino=959815 scontext=system_u:system_r:svirt_lxc_net_t:s0:c808,c926 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:47 ip-10-74-131-153.ec2.internal kernel: audit: type=1400 audit(1463081447.000:10338): avc:  denied  { write } for  pid=15286 comm="sh" path="pipe:[961329]" dev="pipefs" ino=961329 scontext=system_u:system_r:svirt_lxc_net_t:s0:c339,c644 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:47 ip-10-74-131-153.ec2.internal audit[15286]: AVC avc:  denied  { write } for  pid=15286 comm="sh" path="pipe:[961329]" dev="pipefs" ino=961329 scontext=system_u:system_r:svirt_lxc_net_t:s0:c339,c644 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[12842]: AVC avc:  denied  { write } for  pid=12842 comm="java" path="pipe:[477562]" dev="pipefs" ino=477562 scontext=system_u:system_r:svirt_lxc_net_t:s0:c52,c100 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[5511]: AVC avc:  denied  { read } for  pid=5511 comm="java" name="version-2" dev="xvda9" ino=14607984 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=dir permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[5511]: AVC avc:  denied  { read } for  pid=5511 comm="java" name="log.a" dev="xvda9" ino=14608241 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[5511]: AVC avc:  denied  { open } for  pid=5511 comm="java" path="/opt/zookeeper/transactions/version-2/log.a" dev="xvda9" ino=14608241 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[5511]: AVC avc:  denied  { getattr } for  pid=5511 comm="java" path="/opt/zookeeper/transactions/version-2/log.a" dev="xvda9" ino=14608241 scontext=system_u:system_r:svirt_lxc_net_t:s0:c142,c183 tcontext=system_u:object_r:unlabeled_t:s0 tclass=file permissive=1
May 12 19:30:49 ip-10-74-131-153.ec2.internal audit[15641]: AVC avc:  denied  { write } for  pid=15641 comm="entrypoint.sh" path="pipe:[962642]" dev="pipefs" ino=962642 scontext=system_u:system_r:svirt_lxc_net_t:s0:c815,c903 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
May 12 19:30:52 ip-10-74-131-153.ec2.internal audit[15824]: AVC avc:  denied  { write } for  pid=15824 comm="sh" path="pipe:[963328]" dev="pipefs" ino=963328 scontext=system_u:system_r:svirt_lxc_net_t:s0:c522,c523 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=1
```

**Comment by mjg59** [comment]
Which version of CoreOS are you running?

**Comment by anshupande** [comment]
core@ip-10-74-131-153 sandbox-anshu-us-east-1c-control ~ $ cat /etc/os-release
NAME=CoreOS
ID=coreos
VERSION=899.17.0
VERSION_ID=899.17.0
BUILD_ID=2016-05-03-2151
PRETTY_NAME="CoreOS 899.17.0"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"

**Comment by mjg59** [comment]
Ok. Are you able to re-test with the current beta? This will be moving to stable in the near future.

**Comment by anshupande** [comment]
I tried this in beta and did `setenforce 1` the containers stopped with below error:

```May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: auth: no authorization provided, checking guest access
May 20 21:26:58 ip-10-74-131-107.ec2.internal etcd2[1061]: 417cdbb02b580791 [term: 22] ignored a MsgApp message with lower term from fd31e416798d1a53 [term: 21]
^C
core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ cat /etc/os-release
NAME=CoreOS
ID=coreos
VERSION=1010.3.0
VERSION_ID=1010.3.0
BUILD_ID=2016-05-05-0730
PRETTY_NAME="CoreOS 1010.3.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

**Comment by anshupande** [comment]
Also I get these errors: ```core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ journalctl -f|grep avc
May 20 21:45:27 ip-10-74-131-107.ec2.internal audit[18669]: AVC avc:  denied  { write } for  pid=18669 comm="cfn-signal" path="pipe:[173823]" dev="pipefs" ino=173823 scontext=system_u:system_r:svirt_lxc_net_t:s0:c430,c651 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
May 20 21:45:27 ip-10-74-131-107.ec2.internal audit[18669]: AVC avc:  denied  { write } for  pid=18669 comm="cfn-signal" path="pipe:[173824]" dev="pipefs" ino=173824 scontext=system_u:system_r:svirt_lxc_net_t:s0:c430,c651 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
May 20 21:45:31 ip-10-74-131-107.ec2.internal kernel: audit: type=1400 audit(1463780729.633:6713): avc:  denied  { write } for  pid=18776 comm="chronos" path="pipe:[174578]" dev="pipefs" ino=174578 scontext=system_u:system_r:svirt_lxc_net_t:s0:c265,c994 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
May 20 21:45:31 ip-10-74-131-107.ec2.internal kernel: audit: type=1400 audit(1463780729.633:6713): avc:  denied  { write } for  pid=18776 comm="chronos" path="pipe:[174579]" dev="pipefs" ino=174579 scontext=system_u:system_r:svirt_lxc_net_t:s0:c265,c994 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
May 20 21:45:29 ip-10-74-131-107.ec2.internal audit[18776]: AVC avc:  denied  { write } for  pid=18776 comm="chronos" path="pipe:[174578]" dev="pipefs" ino=174578 scontext=system_u:system_r:svirt_lxc_net_t:s0:c265,c994 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
May 20 21:45:29 ip-10-74-131-107.ec2.internal audit[18776]: AVC avc:  denied  { write } for  pid=18776 comm="chronos" path="pipe:[174579]" dev="pipefs" ino=174579 scontext=system_u:system_r:svirt_lxc_net_t:s0:c265,c994 tcontext=system_u:system_r:kernel_t:s0 tclass=fifo_file permissive=0
```

**Comment by anshupande** [comment]
the containers came back but I also get this error: ```core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ docker ps
WARNING: Error loading config file:/home/core/.dockercfg - Invalid Auth config file
```

**Comment by anshupande** [comment]
I did a `core@ip-10-74-131-107 sandbox-ap-us-east-1a-control ~ $ rm /home/core/.dockercfg` to resolve above error ^^

**Comment by mjg59** [comment]
Should be fixed in 1010.5, which is now in stable
